testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestSparkCompatibility.java

```diff
@@ -0,0 +1,104 @@
+
+    @Test(groups = {HIVE_SPARK_BUCKETING, PROFILE_SPECIFIC_TESTS})
+    public void testTrinoReadingTableCreatedByNativeSpark()
+    {
+        // Spark tables can be created using native Spark code or by going through Hive code
+        // This tests the native Spark path.
+        String baseTableName = "test_trino_reading_spark_native_buckets_" + randomTableSuffix();
+
+        String sparkTableDefinitionBase =
+                "CREATE TABLE `default`.`%s` (\n" +
+                        "  `a_string` STRING,\n" +
+                        "  `a_bigint` BIGINT,\n" +
+                        "  `an_integer` INT,\n" +
+                        "  `a_real` FLOAT,\n" +
+                        "  `a_double` DOUBLE,\n" +
+                        "  `a_boolean` BOOLEAN)\n" +
+                        "USING ORC\n" +
+                        "CLUSTERED BY (a_string)\n" +
+                        "INTO 4 BUCKETS\n";
+        String sparkTableDefinition = sparkTableDefinitionBase +
+                        // Hive requires "original" files of transactional tables to conform to the bucketed tables naming pattern
+                        // We can disable transactions or add another pattern to BackgroundHiveSplitLoader
+                        "TBLPROPERTIES ('transactional'='false')";
+        onSpark().executeQuery(format(sparkTableDefinition, baseTableName));
+
+        String values = "VALUES " +
+                "('one', 1000000000000000, 1000000000, 10000000.123, 100000000000.123, true)" +
+                ", ('two', -1000000000000000, -1000000000, -10000000.123, -100000000000.123, false)" +
+                ", ('three', 2000000000000000, 2000000000, 20000000.123, 200000000000.123, true)" +
+                ", ('four', -2000000000000000, -2000000000, -20000000.123, -200000000000.123, false)";
+        String insert = format("INSERT INTO %s %s", baseTableName, values);
+        onSpark().executeQuery(insert);
+
+        Row row1 = row("one", 1000000000000000L, 1000000000, 10000000.123F, 100000000000.123, true);
+        Row row2 = row("two", -1000000000000000L, -1000000000, -10000000.123F, -100000000000.123, false);
+        Row row3 = row("three", 2000000000000000L, 2000000000, 20000000.123F, 200000000000.123, true);
+        Row row4 = row("four", -2000000000000000L, -2000000000, -20000000.123F, -200000000000.123, false);
+
+        String startOfSelect = "SELECT a_string, a_bigint, an_integer, a_real, a_double, a_boolean";
+        QueryResult sparkSelect = onSpark().executeQuery(format("%s FROM %s", startOfSelect, baseTableName));
+        assertThat(sparkSelect).containsOnly(row1, row2, row3, row4);
+
+        QueryResult trinoSelect = onPresto().executeQuery(format("%s FROM %s", startOfSelect, format("%s.default.%s", TRINO_CATALOG, baseTableName)));
+        assertThat(trinoSelect).containsOnly(row1, row2, row3, row4);
+
+        assertThat(onSpark().executeQuery(format("SHOW CREATE TABLE %s", baseTableName)))
```

Comment:
What is this trying to test?

Reply:
It was requested in https://github.com/trinodb/trino/pull/6848#discussion_r583656065
I guess it checks the hidden columns are actually hidden.

Reply:
https://github.com/trinodb/trino/pull/6848#discussion_r583656065 did not request testing `SHOW CREATE TABLE` on _spark_. 

Please remove, as you have SHOW CREATE TABLE test with Trino below